ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -795,6 +801,12 @@ applyByronGenTx _reapply (ByronEBBLedgerConfig (ByronLedgerConfig cfg)) genTx (B
           }
         fixPM (Crypto.AProtocolMagic a b) = Crypto.AProtocolMagic (reAnnotate a) b
 
+mkByronTx :: CC.UTxO.ATxAux ByteString -> GenTx (ByronBlockOrEBB cfg)
+mkByronTx tx = ByronTx
+    { byronTx   = tx
+    , byronTxId = Crypto.hash $ CC.UTxO.taTx tx
```

Comment:
I'm not totally sure if this should hash just the tx body (selected by `taTx`) or also the witnesses. Ask @ruhatch or @dnadales. And I think we should ask the ledger folks to export a `txId` or `txAuxId :: TxAux -> TxId` function from `Cardano.Chain.UTxO.TxAux` that embeds this knowledge (ie what bits to hash, which hash function to use), rather than making us open-code this here.

Reply:
I have created https://github.com/input-output-hk/cardano-ledger/issues/581 for this. I'll add a comment mentioning this issue.